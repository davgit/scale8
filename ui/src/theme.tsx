import { createTheme, adaptV4Theme } from '@mui/material/styles';
import { getProductSection, ProductSectionKey } from './containers/SectionsDetails';
import { grey } from '@mui/material/colors';

declare module '@mui/material/styles/createPalette' {
    interface Palette {
        adminColor: Palette['primary'];
        dataManagerColor: Palette['primary'];
        tagManagerColor: Palette['primary'];
        commonColor: Palette['primary'];
    }
    interface PaletteOptions {
        adminColor: PaletteOptions['primary'];
        dataManagerColor: PaletteOptions['primary'];
        tagManagerColor: PaletteOptions['primary'];
        commonColor: PaletteOptions['primary'];
    }
}

// A custom theme for this app
const theme = createTheme(
    adaptV4Theme({
        palette: {
            primary: {
                main: '#25302e',
            },
            secondary: {
                main: grey['700'],
            },
            error: {
                main: '#c63d51',
            },
            info: {
                main: '#444444',
            },
            success: {
                main: '#c6e54b',
            },
            background: {
                default: '#ffffff',
            },
            adminColor: {
                main: getProductSection(ProductSectionKey.admin).color,
            },
            dataManagerColor: {
                main: getProductSection(ProductSectionKey.dataManager).color,
            },
            tagManagerColor: {
                main: getProductSection(ProductSectionKey.tagManager).color,
            },
            commonColor: {
                main: getProductSection(ProductSectionKey.global).color,
            },
        },
        typography: {
            fontFamily: ['Source Sans Pro', 'Helvetica Neue', 'Arial', 'sans-serif'].join(','),
            fontSize: 14,
            body1: {
                fontSize: 14,
            },
        },
        overrides: {
            MuiTableCell: {
                head: {
                    fontWeight: 600,
                },
            },
            MuiSvgIcon: {
                root: {
                    fontSize: '1.25rem',
                },
            },
            MuiButton: {
                root: {
                    fontWeight: 600,
                },
            },
            MuiTooltip: {
                tooltip: {
                    fontSize: '.8rem',
                },
            },
            MuiDialogActions: {
                spacing: {
                    '& > :not(:first-child)': {
                        marginLeft: '16px',
                    },
                },
            },
        },
    }),
);

export default theme;
