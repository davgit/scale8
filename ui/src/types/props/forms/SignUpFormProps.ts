import { FormProps } from '../../../hooks/form/useFormValidation';
import { ApolloError } from '@apollo/client';
import { RefObject } from 'react';
import HCaptcha from '@hcaptcha/react-hcaptcha';

export type SignUpValues = {
    newPassword: string;
    newPasswordConfirm: string;
    domain: string;
    orgName: string;
    fullName: string;
    email: string;
    agree: boolean;
    CAPTCHAToken: string;
    tempAccessCode: string;
};

export type SignUpFormProps = FormProps<SignUpValues> & {
    gqlError?: ApolloError;
    success: boolean;
    email?: string;
    type: string;
    target: string | undefined;
    loading: boolean;
    qsEmail: string | undefined;
    captcha: RefObject<HCaptcha>;
};
