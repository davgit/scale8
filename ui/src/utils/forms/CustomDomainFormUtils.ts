import { ValidateConfiguration } from '../validators/validateFormValues';
import domainValidator from '../validators/domainValidator';
import { FormProps } from '../../hooks/form/useFormValidation';

export type CustomDomainValues = {
    domain: string;
    certificate: string;
    key: string;
};

export const CustomDomainValidators: ValidateConfiguration<CustomDomainValues>[] = [
    {
        field: 'domain',
        validator: domainValidator,
        error: () => 'Invalid domain',
        ignoreEmpty: true,
    },
];

export type CustomDomainFormProps = FormProps<CustomDomainValues> & {
    cname: string;
};
