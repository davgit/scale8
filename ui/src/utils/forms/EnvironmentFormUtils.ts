import { ValidateConfiguration } from '../validators/validateFormValues';
import nameValidator from '../validators/nameValidator';
import urlValidator from '../validators/urlValidator';

export type EnvVariable = { key: string; value: string };

export type EnvironmentValues = {
    name: string;
    url: string;
    revisionId: string;
    variables?: EnvVariable[];
    comments: string;
};

const EnvironmentCreateValidators: ValidateConfiguration<EnvironmentValues>[] = [
    {
        field: 'name',
        validator: nameValidator,
        error: () => 'Environment name too short',
    },
    {
        field: 'url',
        validator: urlValidator,
        error: () => 'Invalid URL',
        ignoreEmpty: true,
    },
];

const EnvironmentUpdateValidators: ValidateConfiguration<EnvironmentValues>[] = [
    {
        field: 'name',
        validator: nameValidator,
        error: () => 'Environment name too short',
    },
    {
        field: 'url',
        validator: urlValidator,
        error: () => 'Invalid URL',
        ignoreEmpty: true,
    },
];

export { EnvironmentCreateValidators, EnvironmentUpdateValidators };
