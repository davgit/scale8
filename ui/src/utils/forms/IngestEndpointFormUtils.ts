import { ValidateConfiguration } from '../validators/validateFormValues';
import nameValidator from '../validators/nameValidator';
import { StorageProviderFields, storageProviderValidators } from '../StorageProviderUtils';
import { IngestSchemaWizard } from '../../gql/generated/globalTypes';

export type IngestEndpointValues = StorageProviderFields & {
    name: string;
    wizard: IngestSchemaWizard | '';
    analyticsEnabled: boolean;
};

const IngestEndpointValidators: ValidateConfiguration<IngestEndpointValues>[] = [
    {
        field: 'name',
        validator: nameValidator,
        error: () => 'Ingest Endpoint name too short',
    },
    ...storageProviderValidators,
];

export { IngestEndpointValidators };
