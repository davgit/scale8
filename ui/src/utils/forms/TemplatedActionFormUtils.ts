import { PlatformDataMapInput } from '../../types/DataMapsTypes';
import { PlatformActionPermissionInput } from '../../types/ActionPermissionsTypes';
import { ValidateConfiguration } from '../validators/validateFormValues';
import nameValidator from '../validators/nameValidator';
import descriptionValidator from '../validators/descriptionValidator';
import requiredStringValidator from '../validators/requiredStringValidator';
import parsedCodeValidator from '../validators/parsedCodeValidator';
import { PlatformActionPermissionRequest } from '../../gql/generated/globalTypes';

export type TemplatedActionsValues = {
    name: string;
    description: string;
    icon: string;
    code: string;
    execRaw: boolean;
    platformDataMaps: PlatformDataMapInput[];
    permissionRequests: PlatformActionPermissionInput[];
};

const TemplatedActionValidators: ValidateConfiguration<TemplatedActionsValues>[] = [
    {
        field: 'name',
        validator: nameValidator,
        error: () => 'Action name too short',
    },
    {
        field: 'description',
        validator: descriptionValidator,
        error: () => 'Description too short',
    },
    {
        field: 'name',
        validator: requiredStringValidator,
        error: () => 'Required value',
    },
    {
        field: 'code',
        validator: parsedCodeValidator,
        error: (result) => `Invalid Code${result === 0 || `, Error: ${result}`}`,
    },
    {
        field: 'code',
        validator: async (
            value: string | boolean | PlatformActionPermissionInput[] | PlatformDataMapInput[],
        ): Promise<-1 | 0> => {
            return (value as string).includes('success()') ? -1 : 0;
        },
        error: () => 'Action code must invoke "success()"',
    },
    {
        field: 'icon',
        validator: requiredStringValidator,
        error: () => 'Required value',
    },
    {
        field: 'description',
        validator: requiredStringValidator,
        error: () => 'Required value',
    },
    {
        field: 'code',
        validator: requiredStringValidator,
        error: () => 'Required value',
    },
    {
        field: 'permissionRequests',
        validator: async (
            value: string | boolean | PlatformActionPermissionInput[] | PlatformDataMapInput[],
        ): Promise<-1 | 0> => {
            return (value as PlatformActionPermissionInput[]).find((_) => {
                if (
                    _.active &&
                    (_.permission === PlatformActionPermissionRequest.READ_PAGE_URL ||
                        _.permission === PlatformActionPermissionRequest.READ_REFERRER_URL)
                ) {
                    return _.urlParts === undefined || _.urlParts.length === 0;
                }
                return false;
            }) === undefined
                ? -1
                : 0;
        },
        error: () => 'Select at least one.',
    },
    {
        field: 'icon',
        validator: requiredStringValidator,
        error: () => 'Required value',
    },
];

export { TemplatedActionValidators };
