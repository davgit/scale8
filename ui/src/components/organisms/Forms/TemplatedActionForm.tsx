import { ChangeEvent, FC, useEffect, useState } from 'react';
import { Badge, BadgeProps, Box, Button, Divider, lighten } from '@mui/material';
import AppBar from '@mui/material/AppBar';
import Tabs from '@mui/material/Tabs';
import Tab from '@mui/material/Tab';
import TabsTabPanel from '../../molecules/TabsTabPanel';
import { MainDrawerTitle } from '../../molecules/MainDrawerTitle';
import { InfoButton } from '../../molecules/InfoButton';
import { TemplatedActionFormProps } from '../../../types/props/forms/TemplatedActionFormProps';
import ControlledTextInput from '../../atoms/ControlledInputs/ControlledTextInput';
import ControlledTextAreaInput from '../../atoms/ControlledInputs/ControlledTextAreaInput';
import TemplatedActionDataMapSection from '../TemplatedActionDataMapSection';
import TemplatedActionDataMapFormPreview from '../TemplatedActionDataMapFormPreview';
import TemplatedActionPayloadSample from '../TemplatedActionPayloadSample';
import TemplatedActionCodeInput from '../TemplatedActionCodeInput';
import TemplatedActionPermissions from '../TemplatedActionPermissions';
import ControlledSelect from '../../atoms/ControlledInputs/ControlledSelect';
import { getSelectValuesForTypeIcons } from '../../../utils/TypeIconsUtils';
import CheckBoxInput from '../../atoms/InputTypes/CheckBoxInput';
import { useLoggedInState } from '../../../context/AppContext';
import { triggerUpdateTemplatedActionDataMap } from '../../../dialogPages/tagManager/platform/TemplatedActionDataMapUpdate';
import { deleteTemplatedActionDataMap } from '../../../dialogPages/tagManager/platform/TemplatedActionDataMapDelete';
import { triggerAddTemplatedActionDataMap } from '../../../dialogPages/tagManager/platform/TemplatedActionDataMapCreate';
import { styled } from '@mui/material/styles';

const TemplatedActionForm: FC<TemplatedActionFormProps> = (props: TemplatedActionFormProps) => {
    const { gqlError, errors } = props;

    const { templateInteractions } = useLoggedInState();
    const { setSnackbarError, dispatchDialogAction } = templateInteractions;

    useEffect(() => {
        if (gqlError) {
            setSnackbarError(gqlError);
        }
    }, [gqlError]);

    const tabErrors: {
        code: boolean;
        dataMaps: boolean;
        info: boolean;
        permissions: boolean;
    } = {
        info:
            errors.name !== undefined ||
            errors.description !== undefined ||
            errors.icon !== undefined,
        dataMaps: false,
        code: errors.code !== undefined,
        permissions: errors.permissionRequests !== undefined,
    };

    const [tabIndex, setTabIndex] = useState(0);

    const [permissionInfoOpen, setPermissionInfoOpen] = useState(true);

    const handleTabIndexChange = (event: ChangeEvent<unknown>, newValue: number) => {
        setTabIndex(newValue);
    };

    const StyledBadge = styled(Badge)<BadgeProps>(({ theme }) => ({
        '& .MuiBadge-badge': {
            right: -17,
            top: 7,
            backgroundColor: theme.palette.error.main,
        },
    }));

    return (
        <Box
            component="form"
            sx={{
                '& .DrawerFormField': {
                    width: '100%',
                    margin: (theme) => theme.spacing(0, 0, 3),
                },
                minHeight: '100vh',
                display: 'flex',
                flexFlow: 'column nowrap',
            }}
            onSubmit={props.handleSubmit}
        >
            <MainDrawerTitle handleDialogClose={props.handleDialogClose}>
                <Box flexGrow={1}>
                    {props.title}
                    {props.formInfoProps !== undefined && <InfoButton {...props.formInfoProps} />}
                </Box>
                {!props.readOnly && (
                    <Box>
                        <Button
                            size="small"
                            type="submit"
                            sx={{
                                marginRight: (theme) => theme.spacing(2),
                                color: (theme) =>
                                    theme.palette.getContrastText(theme.palette.info.main),
                                backgroundColor: (theme) => theme.palette.info.main,
                                '&:hover': {
                                    backgroundColor: (theme) =>
                                        lighten(theme.palette.info.main, 0.4),
                                },
                            }}
                            disabled={props.isSubmitting}
                        >
                            Save
                        </Button>
                    </Box>
                )}
            </MainDrawerTitle>
            <AppBar position="static" color="default" elevation={0}>
                <Tabs
                    value={tabIndex}
                    onChange={handleTabIndexChange}
                    indicatorColor="primary"
                    textColor="primary"
                    variant="fullWidth"
                >
                    <Tab
                        label={
                            <StyledBadge
                                badgeContent={tabErrors.info ? '!' : undefined}
                                color="secondary"
                            >
                                Info
                            </StyledBadge>
                        }
                        wrapped
                    />
                    <Tab
                        label={
                            <StyledBadge
                                badgeContent={tabErrors.dataMaps ? '!' : undefined}
                                color="secondary"
                            >
                                Form
                            </StyledBadge>
                        }
                        wrapped
                    />
                    <Tab
                        label={
                            <StyledBadge
                                badgeContent={tabErrors.code ? '!' : undefined}
                                color="secondary"
                            >
                                Code
                            </StyledBadge>
                        }
                        wrapped
                    />
                    {!props.values.execRaw && (
                        <Tab
                            label={
                                <StyledBadge
                                    badgeContent={tabErrors.permissions ? '!' : undefined}
                                    color="secondary"
                                >
                                    Permissions
                                </StyledBadge>
                            }
                            wrapped
                        />
                    )}
                </Tabs>
            </AppBar>
            <Divider />
            <Box flex={1} position="relative">
                <Box height="100%" position="absolute" width="100%">
                    <TabsTabPanel value={tabIndex} index={0}>
                        <Box p={3}>
                            <ControlledTextInput
                                name="name"
                                label="Name"
                                formProps={props}
                                className="DrawerFormField"
                                disabled={props.readOnly}
                                requiredOnValidation
                            />
                            <ControlledTextAreaInput
                                name="description"
                                label="Description"
                                formProps={props}
                                className="DrawerFormField"
                                disabled={props.readOnly}
                                requiredOnValidation
                            />
                            <ControlledSelect
                                className="DrawerFormField"
                                label="Icon"
                                name="icon"
                                values={getSelectValuesForTypeIcons('Action')}
                                formProps={props}
                                disabled={props.readOnly}
                                requiredOnValidation
                            />
                            {props.userIsAdmin && (
                                <CheckBoxInput
                                    label="Execute Raw In iFrame"
                                    name="execRaw"
                                    value={props.values['execRaw']}
                                    disabled={props.readOnly || props.isEdit}
                                    setValue={(v) => props.handleChange('execRaw', v)}
                                    className="DrawerFormField"
                                />
                            )}
                        </Box>
                    </TabsTabPanel>
                    <TabsTabPanel value={tabIndex} index={1} fullHeight>
                        <Box display="flex" flexDirection="column" flexWrap="nowrap" height="100%">
                            <Box flex="0 0 50%" height="50%" width="100%" overflow="auto" pb={2}>
                                <TemplatedActionDataMapSection
                                    addDataMap={(parentsIndexes: number[]) => {
                                        triggerAddTemplatedActionDataMap(
                                            parentsIndexes,
                                            props.values.platformDataMaps,
                                            props.handleChange,
                                            dispatchDialogAction,
                                        );
                                    }}
                                    deleteDataMap={(indexes) => {
                                        deleteTemplatedActionDataMap(
                                            indexes,
                                            props.values.platformDataMaps,
                                            props.handleChange,
                                        );
                                    }}
                                    updateDataMap={(indexes: number[]) => {
                                        triggerUpdateTemplatedActionDataMap(
                                            indexes,
                                            props.values.platformDataMaps,
                                            props.handleChange,
                                            dispatchDialogAction,
                                            false,
                                        );
                                    }}
                                    inspectDataMap={(indexes: number[]) => {
                                        triggerUpdateTemplatedActionDataMap(
                                            indexes,
                                            props.values.platformDataMaps,
                                            props.handleChange,
                                            dispatchDialogAction,
                                            true,
                                        );
                                    }}
                                    platformDataMaps={props.values.platformDataMaps}
                                    parentsIndexes={[]}
                                    disabled={props.readOnly}
                                />
                            </Box>
                            <Divider />
                            <Box
                                sx={{
                                    background: `url('/img/pattern.png')`,
                                    padding: (theme) => theme.spacing(1, 2, 1, 2),
                                    fontSize: 24,
                                }}
                            >
                                Form Preview
                            </Box>
                            <Divider />
                            <Box flex="1" height="50%" width="100%" overflow="auto">
                                <Box>
                                    <TemplatedActionDataMapFormPreview
                                        platformDataMaps={props.values.platformDataMaps}
                                    />
                                </Box>
                            </Box>
                        </Box>
                    </TabsTabPanel>
                    <TabsTabPanel value={tabIndex} index={2} fullHeight>
                        <Box display="flex" flexDirection="column" flexWrap="nowrap" height="100%">
                            <Box flex="0 0 50%" width="100%">
                                <TemplatedActionPayloadSample
                                    platformDataMaps={props.values.platformDataMaps}
                                />
                            </Box>
                            <Box flex="0 0 50%" width="100%" p={2}>
                                <TemplatedActionCodeInput {...props} />
                            </Box>
                        </Box>
                    </TabsTabPanel>
                    <TabsTabPanel value={tabIndex} index={3}>
                        <TemplatedActionPermissions
                            {...props}
                            permissionInfoOpen={permissionInfoOpen}
                            setPermissionInfoOpen={setPermissionInfoOpen}
                        />
                    </TabsTabPanel>
                </Box>
            </Box>
        </Box>
    );
};

export default TemplatedActionForm;
