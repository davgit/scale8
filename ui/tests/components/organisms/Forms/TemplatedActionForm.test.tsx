import { render } from '@testing-library/react';
import { ThemeProvider } from '@mui/material';
import theme from '../../../../src/theme';
import { FC } from 'react';
import { useMockFormProps } from '../../../componentsMocks/useMockFormProps';
import TemplatedActionForm from '../../../../src/components/organisms/Forms/TemplatedActionForm';
import { TemplatedActionsValues } from '../../../../src/utils/forms/TemplatedActionFormUtils';

const MockForm: FC = () => {
    const formProps = useMockFormProps<TemplatedActionsValues>({
        name: '',
        description: '',
        icon: '',
        code: '',
        execRaw: false,
        platformDataMaps: [],
        permissionRequests: [],
    });

    return (
        <TemplatedActionForm readOnly={false} isEdit={false} userIsAdmin={true} {...formProps} />
    );
};

describe('TemplatedActionForm', () => {
    afterEach(() => {
        jest.clearAllMocks();
    });
    it('form render unchanged', () => {
        const { asFragment } = render(
            <ThemeProvider theme={theme}>
                <MockForm />
            </ThemeProvider>,
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
