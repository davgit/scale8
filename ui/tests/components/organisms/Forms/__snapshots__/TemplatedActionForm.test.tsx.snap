// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TemplatedActionForm form render unchanged 1`] = `
<DocumentFragment>
  <form
    class="MuiBox-root css-1d8zmdf"
  >
    <div
      class="MuiBox-root css-70qvj9"
    >
      <div
        class="MuiBox-root css-1isemmb"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-erxxaa-MuiSvgIcon-root"
            data-testid="CloseIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiBox-root css-uix0gv"
      >
        <div
          class="MuiBox-root css-i9gxme"
        >
          mock form
        </div>
        <div
          class="MuiBox-root css-0"
        >
          <button
            class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root  css-t6tyzl-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="submit"
          >
            Save
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
    />
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorDefault MuiAppBar-positionStatic css-1w39ges-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiTabs-root css-1ujnqem-MuiTabs-root"
      >
        <div
          class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
          style="overflow: hidden; margin-bottom: 0px;"
        >
          <div
            class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
            role="tablist"
          >
            <button
              aria-selected="true"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary MuiTab-fullWidth MuiTab-wrapped Mui-selected css-otdxzw-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="0"
              type="button"
            >
              <span
                class="BaseBadge-root MuiBadge-root css-1abb7hm-MuiBadge-root"
              >
                Info
                <span
                  class="BaseBadge-badge MuiBadge-badge MuiBadge-standard MuiBadge-invisible MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorSecondary css-1n4jbui-MuiBadge-badge"
                />
              </span>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary MuiTab-fullWidth MuiTab-wrapped css-otdxzw-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="-1"
              type="button"
            >
              <span
                class="BaseBadge-root MuiBadge-root css-1abb7hm-MuiBadge-root"
              >
                Form
                <span
                  class="BaseBadge-badge MuiBadge-badge MuiBadge-standard MuiBadge-invisible MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorSecondary css-1n4jbui-MuiBadge-badge"
                />
              </span>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary MuiTab-fullWidth MuiTab-wrapped css-otdxzw-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="-1"
              type="button"
            >
              <span
                class="BaseBadge-root MuiBadge-root css-1abb7hm-MuiBadge-root"
              >
                Code
                <span
                  class="BaseBadge-badge MuiBadge-badge MuiBadge-standard MuiBadge-invisible MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorSecondary css-1n4jbui-MuiBadge-badge"
                />
              </span>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary MuiTab-fullWidth MuiTab-wrapped css-otdxzw-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="-1"
              type="button"
            >
              <span
                class="BaseBadge-root MuiBadge-root css-1abb7hm-MuiBadge-root"
              >
                Permissions
                <span
                  class="BaseBadge-badge MuiBadge-badge MuiBadge-standard MuiBadge-invisible MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorSecondary css-1n4jbui-MuiBadge-badge"
                />
              </span>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <span
            class="MuiTabs-indicator css-147qviu-MuiTabs-indicator"
            style="left: 0px; width: 0px;"
          />
        </div>
      </div>
    </header>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
    />
    <div
      class="MuiBox-root css-1bqnmih"
    >
      <div
        class="MuiBox-root css-zv8qq2"
      >
        <div
          aria-labelledby="tab-0"
          class="MuiBox-root css-1ufzvhp"
          id="tabpanel-0"
          role="tabpanel"
        >
          <div
            class="MuiBox-root css-19kzrtu"
          >
            <div
              class="MuiFormControl-root MuiTextField-root DialogFormField css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-a6y1oi-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="mui-1"
                id="mui-1-label"
              >
                Name *
              </label>
              <div
                class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-1v2pam7-MuiInputBase-root-MuiInput-root"
              >
                <input
                  aria-invalid="false"
                  autocomplete="off"
                  class="MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-1x51dt5-MuiInputBase-input-MuiInput-input"
                  id="mui-1"
                  name="name"
                  type="text"
                  value=""
                />
                <div
                  class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
                >
                  <button
                    aria-label="toggle password visibility"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-erxxaa-MuiSvgIcon-root"
                      data-testid="ClearIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root DialogFormField css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-ip4mx2-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="mui-2"
                id="mui-2-label"
              >
                Description *
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-multiline css-1yxqjki-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <textarea
                  aria-invalid="false"
                  autocomplete="off"
                  class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                  id="mui-2"
                  name="description"
                  style="height: 0px; overflow: hidden;"
                />
                <textarea
                  aria-hidden="true"
                  class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
                  readonly=""
                  style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding: 0px; width: 100%;"
                  tabindex="-1"
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-1ftyaf0"
                  >
                    <span>
                      Description *
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiFormControl-root DialogFormField css-1nrlq1o-MuiFormControl-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-a6y1oi-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
              >
                Icon *
              </label>
              <div
                class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-foxv06-MuiInputBase-root-MuiInput-root-MuiSelect-root"
              >
                <div
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  aria-labelledby="mui-component-select-icon"
                  class="MuiSelect-select MuiSelect-standard MuiInput-input MuiInputBase-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
                  id="mui-component-select-icon"
                  role="button"
                  tabindex="0"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </div>
                <input
                  aria-hidden="true"
                  class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                  name="icon"
                  tabindex="-1"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-yqkyv6-MuiSvgIcon-root-MuiSelect-icon"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </div>
            </div>
            <label
              class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd DialogFormField css-j204z7-MuiFormControlLabel-root"
            >
              <span
                class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-esq4az-MuiButtonBase-root-MuiCheckbox-root"
              >
                <input
                  class="PrivateSwitchBase-input css-1m9pwf3"
                  data-indeterminate="false"
                  name="execRaw"
                  type="checkbox"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-erxxaa-MuiSvgIcon-root"
                  data-testid="CheckBoxOutlineBlankIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-13p0dqt-MuiTypography-root"
              >
                Execute Raw In iFrame
              </span>
            </label>
          </div>
        </div>
        <div
          aria-labelledby="tab-1"
          class="MuiBox-root css-10klw3m"
          hidden=""
          id="tabpanel-1"
          role="tabpanel"
        />
        <div
          aria-labelledby="tab-2"
          class="MuiBox-root css-10klw3m"
          hidden=""
          id="tabpanel-2"
          role="tabpanel"
        />
        <div
          aria-labelledby="tab-3"
          class="MuiBox-root css-1ufzvhp"
          hidden=""
          id="tabpanel-3"
          role="tabpanel"
        />
      </div>
    </div>
  </form>
</DocumentFragment>
`;
